### ISSUE TITLE
registry_auth::token_not_logged test suite failure

### ISSUE THREAD

**Issue by Apteryks** [rust-lang/cargo#13354]
Hello!

I tried  building rust 1.73.0 and ran its test suite, using GNU Guix.

I expected to see this happen: no failure in test suite

Instead, this happened: `registry_auth::token_not_logged` test failed

### Meta

`rustc --version --verbose`:
```
1.73.0
```

<details><summary>Test output</summary>
<p>

```
failures:

---- registry_auth::token_not_logged stdout ----
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: None,
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/3/b/bar",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/dl/bar/1.0.0/download",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
/tmp/guix-build-rust-1.73.0.drv-0/rustc-1.73.0-src/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/tmp/cit/t2232/dl/bar/1.0.0/download
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/api/v1/crates/new",
        query: None,
        fragment: None,
    },
    method: "put",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/config.json",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
req: Request {
    url: Url {
        scheme: "http",
        cannot_be_a_base: false,
        username: "",
        password: None,
        host: Some(
            Ipv4(
                127.0.0.1,
            ),
        ),
        port: Some(
            44399,
        ),
        path: "/index/3/f/foo",
        query: None,
        fragment: None,
    },
    method: "get",
    authorization: Some(
        "a-unique_token",
    ),
    if_modified_since: None,
    if_none_match: None,
}
thread 'registry_auth::token_not_logged' panicked at tests/testsuite/registry_auth.rs:498:5:
assertion `left == right` failed
  left: 8
 right: 7


failures:
    registry_auth::token_not_logged

test result: FAILED. 2801 passed; 1 failed; 198 ignored; 0 measured; 0 filtered out; finished in 71.12s

error: test failed, to rerun pass `--test testsuite`
```

</p>
</details>

**Comment by weihanglo** [comment]
If I understand correctly, this has been fixed via <https://github.com/rust-lang/cargo/pull/12726>. If possible, could you check test suite in 1.74 or 1.75?

**Comment by weihanglo** [comment]
Going to close as fixed via #12726. Please let us know if the issue is still there.
